Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.Ballistic.cs

```diff
@@ -229,7 +229,7 @@ void SimulateInsertAmmo(EntityUid ammo, EntityUid ammoProvider, EntityCoordinate
         }
 
         List<(EntityUid? Entity, IShootable Shootable)> ammo = new();
-        var evTakeAmmo = new TakeAmmoEvent(20, ammo, Transform(uid).Coordinates, args.User); // RMC14
+        var evTakeAmmo = new TakeAmmoEvent(Math.Min(20, target.Capacity - (target.Entities.Count + target.UnspawnedCount)), ammo, Transform(uid).Coordinates, args.User); // RMC14
```

Comment:
target.Count does the addition for you
also Math.Clamp instead to set a minimum, in case an admin edits it to a silly number and some code freaks out
```suggestion
        var evTakeAmmo = new TakeAmmoEvent(Math.Clamp(target.Capacity - target.Count, 0, 20), ammo, Transform(uid).Coordinates, args.User); // RMC14
```